src/framework/components/element/component.js

```diff
@@ -1668,8 +1674,16 @@ function _define(name) {
         },
         set: function (value) {
             if (this._text) {
+                if (this._text[name] !== value) {
+                    this._dirtyBatch();
```

Comment:
Shouldn't the callback for dirtifying be *after* the value is changed?

Reply:
it does not matter, this just makes it dirty .. the actual value is not used in this function.
And the batch rebuilds after all scripts are done executing.

Reply:
Ahh ok! All good, then. Thanks, @querielo !